ur10_helpers.py

```diff
@@ -0,0 +1,138 @@
+"""Helper functions for controlling UR10 robot arm across different simulation backends."""
+
+import time
+from rclpy.node import Node
+
+
+def move_ur10_joints(node: Node, loop_iteration: int, sim_backend: str):
+    """Publish joint positions for UR10 for the given iteration.
+    
+    Args:
+        node: ROS2 node
+        loop_iteration: Iteration number (0-2) to determine joint positions
+        sim_backend: Simulation backend ("isaacsim", "o3de", or "gazebo")
+    """
+    joint_positions_by_iteration = [
+        [0.0, -1.5708, 1.5708, -1.5708, -1.5708, 0.0],
+        [0.7854, -1.0472, 0.7854, -0.7854, -1.5708, 0.5236],
+        [-0.5236, -2.0944, 2.0944, -2.0944, -1.5708, -0.7854],
+    ]
+    joint_names = [
+        "shoulder_pan_joint",
+        "shoulder_lift_joint",
+        "elbow_joint",
+        "wrist_1_joint",
+        "wrist_2_joint",
+        "wrist_3_joint",
+    ]
+    positions = joint_positions_by_iteration[loop_iteration]
+    
+    if sim_backend == "isaacsim":
+        move_ur10_joints_topic(node, joint_names, positions)
+    elif sim_backend == "o3de":
+        move_ur10_joints_action(node, joint_names, positions)
+    elif sim_backend == "gazebo":
+        move_ur10_joint_array_topic(node, positions)
+    else:
+        print(f"Unknown simulation backend: {sim_backend}")
+
+
+def move_ur10_joints_topic(node: Node, joint_names: list, positions: list):
+    """Move UR10 joints using JointState topic publisher.
+    
+    Used for Isaac Sim backend.
+    
+    Args:
+        node: ROS2 node
+        joint_names: List of joint names
+        positions: List of joint positions (radians)
+    """
+    from sensor_msgs.msg import JointState
+    from std_msgs.msg import Header
+    
+    ur10_joint_pub = node.create_publisher(JointState, "/joint_commands", 10)
+    joint_cmd = JointState()
+    joint_cmd.header = Header()
+    joint_cmd.header.stamp = node.get_clock().now().to_msg()
+    joint_cmd.name = joint_names
+    joint_cmd.position = positions
+    
+    for _ in range(10):
+        ur10_joint_pub.publish(joint_cmd)
+        time.sleep(0.2)
+
+
+def move_ur10_joint_array_topic(node: Node, positions: list):
+    """Move UR10 joints using Float64MultiArray publisher.
+    
+    Used for Gazebo backend.
+    
+    Args:
+        node: ROS2 node
+        positions: List of joint positions (radians)
+    """
+    from std_msgs.msg import Float64MultiArray
+    
+    ur10_joint_pub = node.create_publisher(Float64MultiArray, "/joint_commands", 10)
```

Comment:
suggest to be consistent with the topic from original script

```suggestion
    ur10_joint_pub = node.create_publisher(Float64MultiArray, "/ur10/joint_commands", 10)
```

Reply:
This script does not use namespaces in spawning, hence the robot namespace should be empty.